common/flatpak-dir.c

```diff
@@ -10011,8 +10009,7 @@ _flatpak_dir_fetch_remote_state_metadata_branch (FlatpakDir    *self,
   /* Look up the checksum as advertised by the summary file. If it differs from
    * what we currently have on disk, try and pull the updated ostree-metadata ref.
    * This is how we implement caching. Ignore failure and pull the ref anyway. */
-  if (!flatpak_summary_lookup_ref (state->summary, state->collection_id, OSTREE_REPO_METADATA_REF, &checksum_from_summary, NULL))
-    return flatpak_fail (error, "No such ref (%s, %s) in remote %s", state->collection_id, OSTREE_REPO_METADATA_REF, state->remote_name);
+  flatpak_summary_lookup_ref (state->summary, state->collection_id, OSTREE_REPO_METADATA_REF, &checksum_from_summary, NULL);
```

Comment:
This doesn't look right. state->summary could be NULL here, don't you need to check for that before calling this?
Or maybe use flatpak_remote_state_lookup_ref() which does that check (via flatpak_remote_state_ensure_summary)?

Reply:
I pushed a fixup for this. If we used flatpak_remote_state_lookup_ref() we'd have to clear the error which is a bit messy.

Reply:
This code path should also get more testing once offline installs are working.